### PR TITLE
feat: [EXT-4167][EXT-4168][EXT-4169] Create app analytics client and use it in CCA

### PR DESCRIPTION
**Pull request by loweisz**
Creates the tracking client for cca and app-scripts. 

Also uses the tracking function in the template creation command.

### CODE REVIEW

File: packages/contentful--create-contentful-app/src/index.ts [hunk 1]
```diff
@@ -106,7 +108,13 @@ async function initProject(appName: string, options: CLIOptions) {
 
     console.log(`Creating a Contentful app in ${highlight(tildify(fullAppFolder))}.`);
 
-    await cloneTemplateIn(fullAppFolder, normalizedOptions);
+    const isInteractive = !normalizedOptions.example && !normalizedOptions.source && !normalizedOptions.javascript && !normalizedOptions.typescript
+
+    const templateSource = await getTemplateSource(options)
+
+    track({ template: templateSource, manager: normalizedOptions.npm ? 'npm' : 'yarn', interactive: isInteractive})
```
Comment (colomolo) [inline packages/contentful--create-contentful-app/src/index.ts:115]: Do we want to also track app's (folder) name?

File: packages/contentful--create-contentful-app/src/index.ts [hunk 2]
```diff
@@ -119,6 +127,8 @@ async function initProject(appName: string, options: CLIOptions) {
       await exec('npm', ['install', '--no-audit', '--no-fund'], { cwd: fullAppFolder });
     }
 
+    
+
```
Comment (colomolo) [inline packages/contentful--create-contentful-app/src/index.ts:131]: ✂️

File: packages/contentful--create-contentful-app/src/getTemplateSource.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+function selectTemplate(template: string) {
+  wrapInBlanks(highlight(`---- Cloning the ${chalk.cyan(template)} template...`));
+  return EXAMPLES_PATH + template;
+}
+
+
+async function makeContentfulExampleSource(options: CLIOptions): Promise<string> {
+  if (options.example) {
```
Comment (colomolo) [inline packages/contentful--create-contentful-app/src/getTemplateSource.ts:85]: Maybe refactor all this to `switch`?

File: packages/contentful--create-contentful-app/src/getTemplateSource.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import { highlight, warn, wrapInBlanks } from './logger';
+
+const EXAMPLES_PATH = 'contentful/apps/examples/';
+
+function isContentfulTemplate(url: string) {
+  return Object.values(ContentfulExample).some((t) => url.includes(EXAMPLES_PATH + t));
+}
+
+async function getGithubFolderNames(username: string, repo: string, path: string) {
+  const url = `https://api.github.com/repos/${username}/${repo}/contents/${path}`;
```
Comment (colomolo) [inline packages/contentful--create-contentful-app/src/getTemplateSource.ts:16]: Maybe just hardcode the url here? I think it won't change in the nearest observable future.

File: packages/contentful--app-scripts/utils/analytics.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+const Analytics = require('analytics-node');
```
Comment (vikaskumr) [inline packages/contentful--app-scripts/utils/analytics.js:1]: May be use `import`
```suggestion
import Analytics from 'analytics-node';
```

File: packages/contentful--create-contentful-app/src/index.ts (hunk 1 above)
Comment (loweisz) [reply packages/contentful--create-contentful-app/src/index.ts:115]: To keep the data as anonymous as possible, I would not do that.

File: packages/contentful--create-contentful-app/src/getTemplateSource.ts (hunk 3 above)
Comment (loweisz) [reply packages/contentful--create-contentful-app/src/getTemplateSource.ts:85]: This code was just copy-pasted, but I can change it 👍

File: packages/contentful--create-contentful-app/src/getTemplateSource.ts (hunk 4 above)
Comment (loweisz) [reply packages/contentful--create-contentful-app/src/getTemplateSource.ts:16]: Same as above, was just copy-paste but I can change it

File: packages/contentful--app-scripts/utils/analytics.js (hunk 5 above)
Comment (loweisz) [reply packages/contentful--app-scripts/utils/analytics.js:1]: This project doesn't have a build process, therefore we can not use modules and need to use require

File: packages/contentful--create-contentful-app/src/getTemplateSource.ts (hunk 3 above)
Comment (loweisz) [reply packages/contentful--create-contentful-app/src/getTemplateSource.ts:85]: actually, we are checking here for the existence of different object properties, don't think a switch statement would make much sense here 🤔 Or how would you do that?

Comment (colomolo) [reply packages/contentful--create-contentful-app/src/getTemplateSource.ts:85]: Ah, yeah, sorry, my bad.

**Review by colomolo — APPROVED** [review]
Yeah, looks good to me.

**Pull request merged**